blockdag/dag.go

```diff
@@ -593,6 +594,126 @@ func (dag *BlockDAG) connectBlock(node *blockNode,
 	return chainUpdates, nil
 }
 
+func (node *blockNode) calcMultiset(dag *BlockDAG, transactions []*util.Tx, acceptanceData MultiBlockTxsAcceptanceData, selectedParentUTXO, pastUTXO UTXOSet) (*ecc.Multiset, error) {
+	ms, err := node.pastUTXOMultiSet(dag, acceptanceData, selectedParentUTXO)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, tx := range transactions {
+		ms, err = addTxToMultiset(ms, tx.MsgTx(), pastUTXO, UnacceptedBlueScore)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ms, nil
+}
+
+func (node *blockNode) selectParentMultiSet(dag *BlockDAG) (*ecc.Multiset, error) {
+	if node.isGenesis() {
+		return ecc.NewMultiset(ecc.S256()), nil
+	}
+	return dag.multisetStore.multisetByBlockNode(node.selectedParent)
+}
+
+func (node *blockNode) acceptedSelectedParentMultiset(dag *BlockDAG, acceptanceData MultiBlockTxsAcceptanceData) (*ecc.Multiset, error) {
+	ms, err := node.selectParentMultiSet(dag)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, blockAcceptanceData := range acceptanceData {
+		if !blockAcceptanceData.BlockHash.IsEqual(node.selectedParent.hash) {
```

Comment:
Better find the selectedParent's acceptance data using MultiBlockTxsAcceptanceData.FindAcceptanceData, than run only on it.